File: src/events/fork.rs
use crate::errors::Error;
use crate::events::Event;
use crate::events::base::{Forkee, User, WebhookMessage};
use serde::Deserialize;

#[derive(Deserialize)]
pub struct ForkEvent {
    pub sender: User,
    pub forkee: Forkee,
}

impl Event for ForkEvent {
    fn handle(&self) -> Result<Option<WebhookMessage>, Error> {
        Ok(Some(WebhookMessage {
            content: format!(
                "[{}](<{}>) forked [{}](<{}>)",
                self.sender.login, self.sender.html_url, self.forkee.name, self.forkee.html_url
            ),
            username: self.sender.login.clone(),
            avatar_url: self.sender.avatar_url.clone(),
        }))
    }
}

File: parser/fork.go
package parser

import (
	"fmt"
	"webhook/structs"
)

type fork struct {
	Sender structs.User   `json:"sender"`
	Forkee structs.Forkee `json:"forkee"`
}

func (e *fork) handle() (*structs.Webhook, error) {
	return &structs.Webhook{
		Content: fmt.Sprintf(
			"[%s](<%s>) forked [%s](<%s>)",
			e.Sender.Login,
			e.Sender.HtmlUrl,
			e.Forkee.Name,
			e.Forkee.HtmlUrl,
		),
		Username:  e.Sender.Login,
		AvatarUrl: e.Sender.AvatarURL,
	}, nil
}
